### PR TITLE
Introduce simple test for OM#getDeclaringBean

### LINKED ISSUE #324: Provide test coverage for OM exposing Bean
Issue is described in https://github.com/eclipse-ee4j/cdi/issues/563
If we decide to accept the proposed solution, we'll need a simple TCK test that asserts this for standard class-based bean.

### PR DESCRIPTION
**Pull request by manovotn**
Fixes #324 

First commit is the test addition, second is a small correction to unrelated needed because of generics change in CDI API.

Creating as a draft because this PR will also need a bump in CDI API version.

### CODE REVIEW

File: impl/src/main/java/org/jboss/cdi/tck/tests/event/observer/method/StockWatcher.java [hunk 1]
```diff
@@ -16,10 +16,14 @@
...
 package org.jboss.cdi.tck.tests.event.observer.method;
 
+import jakarta.annotation.Priority;
 import jakarta.enterprise.context.Dependent;
 import jakarta.enterprise.event.Observes;
+import jakarta.enterprise.inject.Alternative;
 
 @Dependent
+@Alternative
+@Priority(1)
```
Comment (Ladicek) [inline impl/src/main/java/org/jboss/cdi/tck/tests/event/observer/method/StockWatcher.java:26]: Are these needed? It was a selected bean before, no?

Comment (Ladicek) [reply impl/src/main/java/org/jboss/cdi/tck/tests/event/observer/method/StockWatcher.java:26]: Ah, it's just to be able to assert more facts about the bean, right?

**Comment by Ladicek** [comment]
Looks fine.

I didn't realize that making the `handles()` return type covariant would require such change, but I believe that's only for people who call `instance.handles().iterator()` directly. If you use foreach, it should be fine.

File: impl/src/main/java/org/jboss/cdi/tck/tests/event/observer/method/StockWatcher.java (hunk 1 above)
Comment (manovotn) [reply impl/src/main/java/org/jboss/cdi/tck/tests/event/observer/method/StockWatcher.java:26]: Exactly, I added that solely to be able to assert a few things about the bean :-)

**Comment by manovotn** [comment]
> I didn't realize that making the handles() return type covariant would require such change, but I believe that's only for people who call instance.handles().iterator() directly. If you use foreach, it should be fine.

Correct, using `for (Instance.Handle<Processor> p : instance.handles()) {...}` works just fine.

**Review by Ladicek — APPROVED** [review]

**Comment by manovotn** [comment]
Updated CDI API.

**Pull request merged**